### ISSUE TITLE
Ignored error in class constructor causes method usage errors to be ignored

### ISSUE THREAD

**Issue by katz12** [microsoft/TypeScript#36268]
**TypeScript Version:**  3.5.3

**Search Terms:**
ts-ignore in class constructor
class method no error on parameter usage

**Code**

```ts
class Foo {
  bar: string
  constructor(bar: string) {
    this.bar = bar
  }

  baz(buzz: number) {
    return buzz + 1
  }
}

// @ts-ignore
const instance = new Foo()
instance.baz('buzz')
```

**Expected behavior:**
The invocation of `instance.baz('buzz')` should report that the parameter should be a `number` instead of a `string`.

**Actual behavior:**
There is no error on that line.
If I fix the ts-ignore on the line above, it then reports the error correctly.

**Playground Link:** 
http://www.typescriptlang.org/play/?ts=3.8.0-dev.20200116#code/MYGwhgzhAEBiD29oG8BQ1oDNEC5oQBcAnASwDsBzdaYeMwogV2APiIApt48HyKBKFNQwEAFiQgA6LtAC8WRNQC+qagCMwHDQC88ZRgFs1AUyKC0GDEWMFGRMtB3QA1NACMy1CtQB6H9AABAggAWhIKMjZjVFp6AmhyQjAyYGM5aDJjAHc4RHZ+VESCZNTJHXYAci4K-iA

**Related Issues:** 
Could not find any through Google or here.

**Comment by DanielRosenwasser** [comment]
This is confusing because from just hovering over `instance`, the type is `Foo`; however, I think what's happening is that we have different type-checker modes, and `new Foo()` produces something different from each type-checker. 

`new Foo()` in our error-producing type-checker returns the internal **error** type that works like `any` because it tries to avoid cascading errors on subsequent operations. However, operations like quick info and auto-completerely on a more lax type-checker that tries to get results of operations in a best-effort way and produces the type `Foo`. However, that more relaxed checker isn't the same type-checker instance that produces errors, so you don't get an error on that subsequent operation.

@sheetalkamat @weswigham does that sound about right?

**Comment by weswigham** [comment]
That is precisely it. See https://github.com/microsoft/TypeScript/pull/28950 for a possible fix.

**Comment by katz12** [comment]
Thanks for the responses. I will focus on fixing type errors in constructors for now.

**Comment by RyanCavanaugh** [comment]
This is fixed now
